### DISCUSSION TITLE
How to define a 2D/3D geometry with holes

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [zwicker-group/py-pde#142] [Q&A] [answered]
Hi,
I was wondering if it's possible to define custom geometries in `py-pde`, for insance a 2D grid with a solid block take out of the domain (see below):
```bash
o o o o o o o o o o
o o o o o o o o o o
o o o x x x x o o o
o o o x x x x o o o
o o o x x x x o o o
```
Thanks!

**Comment by david-zwicker** [comment, marked as answer]
Unfortunately, this is not possible in `py-pde`, since the package is based on simple finite differences. If you are interested in solving PDEs on a more complicated domain, finite volume or finite element methods might be more suitable!

**Reply by ma-sadeghi** [reply]
Thanks for your quick response.
